lib/tasks/csv_report.rake

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require "csv"
+
+namespace :csv_report do
+  desc "A CSV report of political documents published by organisations"
+  task political_status_by_organisation: :environment do
+    raise "csv_report:political_status_by_organisation rake task should be removed" unless Time.zone.today < Date.new(2020, 1, 1)
+
+    # Get all primary publishing organisations in use
+    org_content_ids = Edition.where(live: true).joins(revision: :tags_revision).pluck(Arel.sql("tags_revisions.tags->'primary_publishing_organisation'->0")).uniq
+
+    links = Linkables.new("organisation")
+
+    organisations = org_content_ids.map { |id| links.by_content_id(id) }.compact
+
+    organisations.each do |organisation|
+      path = Rails.root.join("tmp", "#{organisation['internal_name'].parameterize}-political-status.csv")
+      csv_headers = ["Public URL", "Admin URL", "Title", "Document type", "Political", "Summary"]
+
+      CSV.open(path, "w", headers: csv_headers, write_headers: true) do |csv|
+        editions_published_by_organisation = Edition.where(live: true).joins(revision: :tags_revision).merge(TagsRevision.primary_organisation_is(organisation["content_id"]))
```

Comment:
`Edition.where(live: true).joins(revision: :tags_revision)` is a repetition of line 11 - I would be tempted to move this out into a variable on line 9: `live_editions = Edition.where(live: true).joins(revision: :tags_revision)`

Can then reuse this value on line 11 and 22, saving a bit of database querying (and shortening the lines, not as much chaining needed).